src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/ReportService.kt

```diff
@@ -195,4 +197,15 @@ class ReportService(
         WorkbookFactory.create(true)
       }
   }
+
+  private fun splitAndRetrievePersonInfo(crns: Set<String>): Map<String, PersonSummaryInfoResult> {
+    val deliusUsername = userService.getUserForRequest().deliusUsername
+
+    val crnMap = ListUtils.partition(crns.toList(), numberOfCrn)
+      .stream().map { crns ->
+        offenderService.getOffenderSummariesByCrns(crns.toSet(), deliusUsername).associateBy { it.crn }
```

Comment:
If 500 is the hard limit for this API, wouldn't it make more sense to do this partitioning in ApDeliusContextApiOffenderDetailsDataSource.getOffenderDetailSummaries? This will transparently protect us from this issue in other places and i think would mostly just be a case of shifting this partition logic into that method

Reply:
Its a good idea, i want to try this fix to see if it resolve the current problem in reporting. 

Most of the cases we call this endpoint with single crn, only couple of places we called with more than 500 crns
I would create card to refactor this logic into common place once it resolves the problem.

Reply:
Sure, could you please create that now and link here. ta.

Reply:
actually don't worry let's see it fixed first. will approve

Reply:
Card to refactor this logic in common client level after we confident with the solution: https://dsdmoj.atlassian.net/browse/CAS-246